pkg/admission/validator_test.go

```diff
@@ -0,0 +1,229 @@
+	}
+}
+
+func TestSAValidator_ValidateHTTPRoute(t *testing.T) {
+	testService := "test-service"
+	specialService := "special-service"
+	tests := []struct {
+		name        string
+		hRoute      sav1.HTTPRoute
+		wantOK      bool
+		wantMessage string
+		wantErr     bool
+	}{
+		{
+			name: "valid httpRoute with no filters",
+			hRoute: sav1.HTTPRoute{
+				Spec: sav1.HTTPRouteSpec{
+					Rules: []sav1.HTTPRouteRule{
+						{
+							Matches: []sav1.HTTPRouteMatch{
+								{
+									Path: sav1.HTTPPathMatch{
+										Type:  "Prefix",
+										Value: "/",
+									},
+								},
+							},
+							ForwardTo: []sav1.HTTPRouteForwardTo{
+								{
+									ServiceName: &testService,
+									Port: 8080,
+									Weight: 100,
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOK:      true,
+			wantMessage: "",
+			wantErr:     false,
+		},
+		{
+			name: "valid httpRoute with 1 filter",
+			hRoute: sav1.HTTPRoute{
+				Spec: sav1.HTTPRouteSpec{
+					Rules: []sav1.HTTPRouteRule{
+						{
+							Matches: []sav1.HTTPRouteMatch{
+								{
+									Path: sav1.HTTPPathMatch{
+										Type:  "Prefix",
+										Value: "/",
+									},
+								},
+							},
+							Filters: []sav1.HTTPRouteFilter{
+								{
+									Type: "RequestMirror",
+									RequestMirror: &sav1.HTTPRequestMirrorFilter{
+										ServiceName: &testService,
+										Port:        8080,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOK:      true,
+			wantMessage: "",
+			wantErr:     false,
+		},
+		{
+			name: "valid httpRoute with 2 filters",
+			hRoute: sav1.HTTPRoute{
+				Spec: sav1.HTTPRouteSpec{
+					Rules: []sav1.HTTPRouteRule{
+						{
+							Matches: []sav1.HTTPRouteMatch{
+								{
+									Path: sav1.HTTPPathMatch{
+										Type:  "Prefix",
+										Value: "/",
+									},
+								},
+							},
+							Filters: []sav1.HTTPRouteFilter{
+								{
+									Type: "RequestMirror",
+									RequestMirror: &sav1.HTTPRequestMirrorFilter{
+										ServiceName: &testService,
+										Port:        8080,
+									},
+								},
+								{
+									Type: "RequestMirror",
+									RequestMirror: &sav1.HTTPRequestMirrorFilter{
+										ServiceName: &specialService,
+										Port:        8081,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOK:      false,
+			wantMessage: "HTTPRules cannot contain more than one HTTPFilter",
+			wantErr:     false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1, err := ValidateHTTPRoute(tt.hRoute)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("KongHTTPValidator.ValidateCredential() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.wantOK {
+				t.Errorf("KongHTTPValidator.ValidateCredential() got = %v, want %v", got, tt.wantOK)
+			}
+			if got1 != tt.wantMessage {
+				t.Errorf("KongHTTPValidator.ValidateCredential() got1 = %v, want %v", got1, tt.wantMessage)
```

Comment:
What is the significance of the Kong reference here?

Reply:
Sigh, this was an unfortunate side effect of borrowing the struct layout of a Kong server test to play with some of it’s tests. I’m not sure if we actually need all of these tests but will change them to the correct values if I keep them.